src/librustc_trans/trans/mir/lvalue.rs

```diff
@@ -0,0 +1,149 @@
+    {
+        let lltemp = base::alloc_ty(bcx, ty, name);
+        LvalueRef::new(lltemp, LvalueTy::from_ty(ty))
+    }
+}
+
+impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
+    pub fn trans_lvalue(&self,
+                        bcx: Block<'bcx, 'tcx>,
+                        lvalue: &mir::Lvalue<'tcx>)
+                        -> LvalueRef<'tcx> {
+        debug!("trans_lvalue(lvalue={:?})", lvalue);
+
+        let fcx = bcx.fcx;
+        let ccx = fcx.ccx;
+        let tcx = bcx.tcx();
+        match *lvalue {
+            mir::Lvalue::Var(index) => self.vars[index as usize],
+            mir::Lvalue::Temp(index) => self.temps[index as usize],
+            mir::Lvalue::Arg(index) => self.args[index as usize],
+            mir::Lvalue::Static(_def_id) => panic!("NYI -- statics"),
+            mir::Lvalue::ReturnPointer => {
+                let llval = fcx.llretslotptr.get().unwrap();
+                let return_ty = bcx.monomorphize(&self.mir.return_ty.unwrap());
+                LvalueRef::new(llval, LvalueTy::from_ty(return_ty))
+            }
+            mir::Lvalue::Projection(ref projection) => {
+                let tr_base = self.trans_lvalue(bcx, &projection.base);
+                let projected_ty = tr_base.ty.projection_ty(tcx, &projection.elem);
+                let llprojected = match projection.elem {
+                    mir::ProjectionElem::Deref => {
+                        let base_ty = tr_base.ty.to_ty(tcx);
+                        base::load_ty(bcx, tr_base.llval, base_ty)
+                    }
+                    mir::ProjectionElem::Field(ref field) => {
+                        let base_ty = tr_base.ty.to_ty(tcx);
+                        let base_repr = adt::represent_type(ccx, base_ty);
+                        let discr = match tr_base.ty {
+                            LvalueTy::Ty { .. } => 0,
+                            LvalueTy::Downcast { adt_def: _, substs: _, variant_index: v } => v,
+                        };
+                        let discr = discr as u64;
+                        adt::trans_field_ptr(bcx, &base_repr, tr_base.llval, discr, field.index())
+                    }
+                    mir::ProjectionElem::Index(ref index) => {
+                        let base_ty = tr_base.ty.to_ty(tcx);
+                        let index = self.trans_operand(bcx, index);
+                        let llindex = self.prepare_index(bcx, index.llval);
+                        let (llbase, _) = tvec::get_base_and_len(bcx, tr_base.llval, base_ty);
+                        build::InBoundsGEP(bcx, llbase, &[llindex])
+                    }
+                    mir::ProjectionElem::ConstantIndex { offset,
+                                                         from_end: false,
+                                                         min_length: _ } => {
+                        let base_ty = tr_base.ty.to_ty(tcx);
+                        let lloffset = common::C_u32(bcx.ccx(), offset);
```

Comment:
Hmm, thinking about it, that would make the available values differ when cross-compiling. 32bit at least gives consistent results for now.